# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1758366728 -7200
# Parent  58f73980de151e6ce39dd84b912912cc6228499d
Misc: Add more junk controls to Junk button on unified toolbar.

diff --git a/mail/components/unifiedtoolbar/content/unifiedToolbarCustomizableItems.inc.xhtml b/mail/components/unifiedtoolbar/content/unifiedToolbarCustomizableItems.inc.xhtml
--- a/mail/components/unifiedtoolbar/content/unifiedToolbarCustomizableItems.inc.xhtml
+++ b/mail/components/unifiedtoolbar/content/unifiedToolbarCustomizableItems.inc.xhtml
@@ -149,9 +149,9 @@
 
 <html:template id="junkTemplate"
                xmlns="http://www.w3.org/1999/xhtml">
   <button is="mail-tab-button"
-          command="cmd_markAsJunk"
+          popup="toolbarJunkPopup"
           label-id="toolbar-junk-label"
           data-l10n-id="toolbar-junk"></button>
 </html:template>
 
diff --git a/mail/components/unifiedtoolbar/content/unifiedToolbarPopups.inc.xhtml b/mail/components/unifiedtoolbar/content/unifiedToolbarPopups.inc.xhtml
--- a/mail/components/unifiedtoolbar/content/unifiedToolbarPopups.inc.xhtml
+++ b/mail/components/unifiedtoolbar/content/unifiedToolbarPopups.inc.xhtml
@@ -93,8 +93,22 @@
             accesskey="&markStarredCmd.accesskey;"
             key="key_toggleFlagged"
             command="cmd_markAsFlagged"/>
 </menupopup>
+<menupopup id="toolbarJunkPopup" onpopupshowing="document.commandDispatcher.updateCommands('create-menu-mark')">
+  <menuitem id="button-markAsJunk"
+            label="&markAsJunkCmd.label;"
+            accesskey="&markAsJunkCmd.accesskey;"
+            command="cmd_markAsJunk"/>
+  <menuitem id="button-markAsNotJunk"
+            label="&markAsNotJunkCmd.label;"
+            accesskey="&markAsNotJunkCmd.accesskey;"
+            command="cmd_markAsNotJunk"/>
+  <menuitem id="button-recalculateJunkScore"
+            label="&recalculateJunkScoreCmd.label;"
+            accesskey="&recalculateJunkScoreCmd.accesskey;"
+            command="cmd_recalculateJunkScore"/>
+</menupopup>
 <menupopup id="toolbarViewPickerPopup"
             onpopupshowing="RefreshViewPopup(this);">
   <menuitem id="viewPickerAll" value="0"
             label="&viewAll.label;"
